### PR TITLE
[gha] Add GitHub Action workflow to automate tests

### PR DESCRIPTION
**Pull request by jjmerchante**
This PR adds a new GitHub action workflow. It is triggered whenever there is a push or a pull
request event to the unicorn branch. It runs the tests for both backend and fronted with different Python
and node versions.

Node 13.x isn't compatible with a package. But also node 13.x is not maintained anymore as it is not LTS, so I disabled it.

It also updates UI files to pass yarn lint tests (some of the files are automatically updated by yarn lint).

Issue related: https://github.com/chaoss/grimoirelab-bestiary/issues/86

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
+name: tests
+
+on:
+  push:
+    branches:
+      - unicorn
+    tags:
+      - '!*.*.*'
```
Comment (sduenas) [inline .github/workflows/tests.yml]:
```suggestion
      - '!**''
```

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install and set up Poetry
+      run: curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python
+    - name: Install dependencies
+      run: |
+        source $HOME/.poetry/env
+        poetry install -vvv
```
Comment (sduenas) [inline .github/workflows/tests.yml]: It's better to do this during poetry setup:

https://github.com/chaoss/grimoirelab-perceval/blob/master/.github/workflows/tests.yml#L30

**Review by sduenas — CHANGES_REQUESTED** [review]
Please check my comments.

**Comment by jjmerchante** [comment]
Do I also include coverage analysis?

File: pyproject.toml [hunk 3]
```diff
@@ -15,6 +15,7 @@ graphene-django = "^2"
 django-graphql-jwt = "^0.3.0"
 PyJWT = "^1.7.0"
 django-cors-headers = "^3.7.0"
+flake8 = "^3.9.2"
```
Comment (vchrombie) [inline pyproject.toml]: Do you think we can move `flake8` to the dev-dependencies?

We usually keep it there in other grimoirelab repositories.
https://github.com/chaoss/grimoirelab-perceval/blob/master/pyproject.toml#L55

**Review by vchrombie — COMMENTED** [review]
I just had one minor comment.

**Comment by vchrombie** [comment]
@jjmerchante 
Also, a badge on the `unicorn` README.md would be good. WDYT?

# Bestiary [Build Status](https://github.com/chaoss/grimoirelab-bestiary/actions?query=workflow:tests+branch:unicorn+event:push)

```
# Bestiary [![Build Status](https://github.com/chaoss/grimoirelab-bestiary/workflows/tests/badge.svg?branch=unicorn)](https://github.com/chaoss/grimoirelab-bestiary/actions?query=workflow:tests+branch:unicorn+event:push)
```

Ref: https://github.com/chaoss/grimoirelab-sortinghat/tree/muggle#sorting-hat--

Sorry for not reporting this earlier. 😅

**Review by sduenas — APPROVED** [review]
LGTM, @vchrombie, what do you think?

**Review by vchrombie — APPROVED** [review]
LGTM!
Thanks for working on the comments @jjmerchante. 🚀

**Pull request merged**